### PR TITLE
fix select broker is none

### PR DESCRIPTION
**Pull request by lordcheng10**
### Motivation
In selectBrokerForAssignment, when the broker is selected again, it is not judged whether the selected broker is empty.

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Review by gaozhangmin — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java [hunk 1]
```diff
@@ -878,7 +878,10 @@ public Optional<String> selectBrokerForAssignment(final ServiceUnitId serviceUni
                     LoadManagerShared.applyNamespacePolicies(serviceUnit, policies, brokerCandidateCache,
                             getAvailableBrokers(),
                             brokerTopicLoadingPredicate);
-                    broker = placementStrategy.selectBroker(brokerCandidateCache, data, loadData, conf);
+                    Optional<String> brokerTmp = placementStrategy.selectBroker(brokerCandidateCache, data, loadData, conf);
```
Comment (gaozhangmin) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java:881]: `src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java:[881] (sizes) LineLength: Line is longer than 120 characters (found 124).`
Please fix checkstyle violation

Comment (lordcheng10) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java]: Fixed.

**Comment by heesung-sn** [comment]
LGTM

**Review by merlimat — APPROVED** [review]

**Pull request merged**